src/main/resources/errors.yaml

```diff
@@ -134,6 +134,95 @@ request-errors:
     body: |-
       big long message with ${vars} in it
 
+  # Schema Errors
+
+  - scope: SCHEMA
+    code: COLUMN_DEFINITION_MISSING
+    title: Column definition is missing for the provided primary key column.
+    body: |-
+      Column definition is missing for the provided primary key column: ${column_name}.
+
+  - scope: SCHEMA
+    code: COLUMN_TYPE_INCORRECT
+    title: Column data type not provided or format invalid in the definition
+    body: |-
+      Column data type not provided or format invalid in the definition.
+      Column definition can be defined in shorthand format as : 
+        "column_name": "text"
+      
+      or in nested object structure format as:
+        "column_name": {
+          "type": "text"
+        }
+
+  - scope: SCHEMA
+    code: COLUMN_TYPE_UNSUPPORTED
+    title: Column data type used in the definition is not supported.
+    body: |-
+      Column data type ${type} used in the definition is not supported.
+      Following are the supported data types: ${supported_types}
+
+  - scope: SCHEMA
+    code: MISSING_PRIMARY_KEYS
+    title: Primary key definition needs to be provided while creating the table.
+    body: |-
+      Primary key definition needs to be provided while creating the table.
```

Comment:
Hmmh. This looks like a very long multi-line exception message. Is this something we should be doing? I guess it should be discussed with @amorton but to me these can be noisy in logs and seems like we should have a link instead of novel-length tutorial of usage. :)

Reply:
Had the same feel, raised it in the discussion yesterday, this is the messaging approach for error, that's planed. cc: @amorton